sockets.c

```diff
@@ -31,933 +33,598 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-/* network stub calls */
 #include <uk/config.h>
-#include <sys/time.h>
-#if CONFIG_LWIP_SOCKET_PPOLL
-#include <signal.h>
-#endif
-#include <vfscore/dentry.h>
-#include <vfscore/file.h>
-#include <vfscore/fs.h>
-#include <vfscore/mount.h>
-#include <vfscore/vnode.h>
-#include <uk/alloc.h>
+#include <uk/assert.h>
+#include <uk/print.h>
 #include <uk/essentials.h>
-#include <uk/errptr.h>
-#include <stdio.h>
-#include <errno.h>
-#include <lwip/sockets.h>
-#include <lwip/if_api.h>
+#include <uk/socket_driver.h>
+#include <vfscore/eventpoll.h>
 
-#define SOCK_NET_SET_ERRNO(errcode) \
-	(errno = -(errcode))
-
-static int sock_net_close(struct vnode *s_vnode,
-			struct vfscore_file *vfscore_file);
-static int sock_net_write(struct vnode *s_vnode,
-			struct uio *buf, int ioflag __unused);
-static int sock_net_read(struct vnode *s_vnode,
-			struct vfscore_file *vfscore_file __unused,
-			struct uio *buf, int ioflag __unused);
-static int sock_net_ioctl(struct vnode *s_vnode,
-			struct vfscore_file *vfscore_file __unused,
-			unsigned long request,
-			void *buf);
-
-#define sock_net_getattr   ((vnop_getattr_t) vfscore_vop_einval)
-#define sock_net_inactive  ((vnop_inactive_t) vfscore_vop_nullop)
-
-static struct vnops sock_net_vnops = {
-	.vop_close = sock_net_close,
-	.vop_write = sock_net_write,
-	.vop_read  = sock_net_read,
-	.vop_ioctl = sock_net_ioctl,
-	.vop_getattr = sock_net_getattr,
-	.vop_inactive = sock_net_inactive
+#include <lwip/sockets.h>
+#include <lwip/priv/sockets_priv.h>
+#include <lwip/api.h>
+#include <lwip/sys.h>
+
+struct lwip_socket_data {
+	/* fd of the corresponding lwip socket */
+	int lwip_fd;
+
+	/* List of registered eventpolls. The list is synchronized with
+	 * lwip SYS_ARCH_PROTECT, as this lock is held anyways during the event
+	 * callback and needed during poll to receive a current event state
+	 * from the lwip socket.
+	 */
+	struct uk_list_head ep_list;
```

Comment:
This is just a personal preference, but I think `evp_list` is a bit more intuitive. I struggled a bit with `ep_list` while browsing the code.